File: lectures/week-1/exercisses/calculator.c
#include <stdio.h>

float add(float num1, float num2);
float subtract(float num1, float num2);
float multiply(float num1, float num2);
float divide(float num1, float num2);

int main(void)
{
    float num1, num2;

    printf("Enter the first number: ");
    scanf("%f", &num1);

    printf("Enter the second number: ");
    scanf("%f", &num2);

    printf("%.2f + %.2f = %.2f\n", num1, num2, add(num1, num2));
    printf("%.2f - %.2f = %.2f\n", num1, num2, subtract(num1, num2));
    printf("%.2f * %.2f = %.2f\n", num1, num2, multiply(num1, num2));
    printf("%.2f / %.2f = %.2f\n", num1, num2, divide(num1, num2));
}

float add(float num1, float num2)
{
    return num1 + num2;
}

float subtract(float num1, float num2)
{
    return num1 - num2;
}

float multiply(float num1, float num2)
{
    return num1 * num2;
}

float divide(float num1, float num2)
{
    return num1 / num2;
}

File: lectures/week-6/exercises/calculator.py
def add(x, y) -> float:
    """Adds two numbers."""
    return round(float(x) + float(y), 2)

def subtract(x, y) -> float:
    """Subtracts two numbers."""
    return round(float(x) - float(y), 2)

def multiply(x, y) -> float:
    """Multiplies two numbers."""
    return round(float(x) * float(y), 2)

def divide(x, y) -> float:
    """Divides two numbers, handling division by zero."""
    try:
        return round(float(x) / float(y), 2)
    except ZeroDivisionError:
        return "Division by zero is not allowed."

x = input("Enter a number: ")
y = input("Enter another number: ")

print(f"{x} + {y} = {add(x, y)}")
print(f"{x} - {y} = {subtract(x, y)}")
print(f"{x} * {y} = {multiply(x, y)}")
print(f"{x} / {y} = {divide(x, y)}")
